### PR TITLE
Create BlockPathRegex Middleware

### LINKED ISSUE #4813: Create BlockPathRegex Middleware
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

Since golang and goregexp does not support lookaheads and readbehinds, issues come into play when we want to expose all except a specific path. For example, `PathPrefix: /`, but NOT `/bacon`.

There should be configurable response codes for the block match.

By having this coded in middlewares, it could be re-used, and shared between providers.

### PR DESCRIPTION
**Pull request by dtomcej**
### What does this PR do?

Creates a regex middleware that blocks requests, and returns a configured response code and message.

### Motivation

Fixes #4813, #4812 

### More

- [x] Added/updated tests
- [ ] Added/updated documentation

### CODE REVIEW

File: pkg/config/middlewares.go [hunk 1]
```diff
@@ -327,6 +327,15 @@ type ReplacePathRegex struct {
 
 // +k8s:deepcopy-gen=true
 
+// BlockPathRegex holds the BlockPathRegex configuration.
+type BlockPathRegex struct {
+	Regex        string `json:"regex,omitempty"`
+	ResponseCode int    `json:"responseCode,omitempty"`
+	Message      string `json:"message,omitempty"`
+}
```
Comment (sylr) [inline pkg/config/middlewares.go]: Shouldn't `BlockPathRegex` be added to the `Middleware` struct ?

**Comment by sylr** [comment]
@dtomcej are you still working on this ?

**Comment by ldez** [comment]
This PR in in **status/1-needs-design-review**.

https://docs.traefik.io/contributing/maintainers/#pr-review-process

**Comment by sylr** [comment]
@ldez it has been in this state for 7 months now. Can someone do the design review ?

**Comment by ldez** [comment]
We consider that this middleware is a routing constraint, and we think that this kind must be able by the routing itself and not a middleware.

So I close for now.

**Pull request closed without merging**